src/common/core/users/metarename.js

```
@@ -0,0 +1,358 @@
+                    definitionInfo = core.getPointerDefinitionInfo(visited, parameters.newName, target);
+                    if (definitionInfo.sourcePath === nodePath &&
+                        definitionInfo.targetPath === parameters.targetPath) {
+                        core.renamePointer(visited, parameters.oldName, parameters.newName);
+                    }
+                    deferred.resolve(null);
+                })
+                .catch(function (err) {
+                    deferred.resolve(null);
+                });
+
+            return deferred.promise.nodeify(next);
+        }
+
+        function visitForSet(visited, next) {
+            var definitionInfo,
+                deferred = Q.defer();
+
+            if (parameters.excludeOriginNode === true && core.getPath(visited) === nodePath) {
+                return Q.resolve(null).nodeify(next);
+            }
+
+            if (core.getValidSetNames(visited).indexOf(parameters.newName) === -1 ||
+                core.getOwnSetNames(visited, parameters.oldName).indexOf(parameters.oldName) === -1) {
+                return Q.resolve(null).nodeify(next);
+            }
+
+            core.loadMembers(visited, parameters.oldName)
+                .then(function (members) {
+                    var i;
+
+                    for (i = 0; i < members.length; i += 1) {
+                        definitionInfo = core.getPointerDefinitionInfo(visited, parameters.newName, members[i]);
+                        if (definitionInfo.sourcePath === nodePath &&
+                            definitionInfo.targetPath === parameters.targetPath) {
+                            core.moveMember(visited, core.getPath(members[i]), parameters.oldName, parameters.newName);
+                        }
+                    }
+                    deferred.resolve(null);
+                    return;
+                })
+                .catch(function (err) {
+                    deferred.resolve(null);
```

Should all errors be swallowed here?